### PR TITLE
Postgres: Allow negative integers in sequences

### LINKED ISSUE #5908: Postgres: failure to parse negative sequence increments
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried linting a `CREATE SEQUENCE .. INCREMENT BY X` statement where `X` is `-1`.

It fails to parse with

```
L:   1 | P:  22 |  PRS | Line 1, Position 22: Found unparsable section:
                       | 'increment by -1;'
```

### Expected Behaviour

I would expect linting to succeed, negative increments are [a documented feature](https://www.postgresql.org/docs/13/sql-createsequence.html) and are valid.

### Observed Behaviour

SQLFluff fails to parse, while it works for positive increments.

### How to reproduce

1. put code in file
```
create sequence derp increment by -5;
```
2. run `sqlfluff lint --dialect=postgres myfile.sql`
3. observe failure

### Dialect

Postgres

### Version

3.0.6

### Configuration

[sqlfluff]
dialect = postgres
templater = raw
max_line_length = 120
large_file_skip_byte_limit = 0
exclude_rules = jinja
processes = -1

[sqlfluff:indentation]
allow_implicit_indents = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:convention.not_equal]
preferred_not_equal_style = c_style

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by WittierDinosaur**
### Brief summary of the change made
Fixes #5908

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_postgres.py [hunk 1]
```diff
@@ -4178,18 +4178,32 @@ class CreateSequenceOptionsSegment(ansi.CreateSequenceOptionsSegment):
     match_grammar = OneOf(
         Sequence("AS", Ref("DatatypeSegment")),
         Sequence(
-            "INCREMENT", Ref.keyword("BY", optional=True), Ref("NumericLiteralSegment")
+            "INCREMENT",
+            Ref.keyword("BY", optional=True),
+            Ref("NegativeSegment", optional=True),
```
Comment (keraion) [inline src/sqlfluff/dialects/dialect_postgres.py:4183]: The `NegativeSegment` should be replaced with `SignedSegmentGrammar` as values like `+4` are also allowed.

**Review by keraion — CHANGES_REQUESTED** [review]
I'd suggest changing all the `NegativeSegment`s with `SignedSegmentGrammar`.

**Comment by WittierDinosaur** [comment]
@keraion addressed 👍

**Review by keraion — APPROVED** [review]
LGTM

**Pull request merged**